boefjes/boefjes/plugins/kat_servicebanner/main.py

```diff
@@ -0,0 +1,40 @@
+import socket
+
+from boefjes.job_models import BoefjeMeta
+
+TIMEOUT = 1.0
+
+
+def get_ssh_sock(ip, port, timeout):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.settimeout(timeout)
+    try:
+        sock.connect((ip, port))
+        return sock
+    except Exception:
+        return None
+
+
+def get_ssh_banner(sock):
+    try:
+        banner = sock.recv(1024)
```

Comment:
Maybe add another timeout call for this read? I'm not sure the initial timeout you set also applies to slow reads, since it was set before the connect call.